templates/primary-store-service.json

```diff
@@ -0,0 +1,417 @@
+         "metadata": {
+            "name": "${APPLICATION_NAME}"
+         },
+         "stringData" : {
+            "application-user" : "${APPLICATION_USER}",
+            "application-password" : "${APPLICATION_USER_PASSWORD}"
+         }
+      },
+      {
+         "kind": "RoleBinding",
+         "apiVersion": "v1",
+         "metadata": {
+            "name": "${APPLICATION_NAME}-view"
+         },
+         "userNames": [
+            "system:serviceaccount:${NAMESPACE}:${APPLICATION_NAME}"
+         ],
+         "groupNames": null,
+         "subjects": [
+            {
+               "kind": "ServiceAccount",
+               "name": "${APPLICATION_NAME}"
+            }
+         ],
+         "roleRef": {
+            "name": "view"
+         }
+      },
+      {
+         "apiVersion": "v1",
+         "kind": "Service",
+         "metadata": {
+            "annotations": {
+               "description": "Headless service for StatefulSets"
+            },
+            "labels": {
+               "application": "${APPLICATION_NAME}"
+            },
+            "name": "${APPLICATION_NAME}-headless"
+         },
+         "spec": {
+            "ports": [
+               {
+                  "name": "http",
+                  "port": 8080,
+                  "targetPort": 8080
+               },
+               {
+                  "name": "hotrod",
+                  "port": 11222,
+                  "targetPort": 11222
+               }
+            ],
+            "selector": {
+               "deploymentConfig": "${APPLICATION_NAME}"
+            },
+            "clusterIP": "None"
+         }
+      },
+      {
+         "apiVersion": "v1",
+         "kind": "Service",
+         "metadata": {
+            "annotations": {
+               "description": "The web server's HTTP port."
+            },
+            "labels": {
+               "application": "${APPLICATION_NAME}"
+            },
+            "name": "${APPLICATION_NAME}-http"
+         },
+         "spec": {
+            "ports": [
+               {
+                  "port": 8080,
+                  "targetPort": 8080
+               }
+            ],
+            "selector": {
+               "deploymentConfig": "${APPLICATION_NAME}"
+            }
+         }
+      },
+      {
+         "apiVersion": "v1",
+         "kind": "Service",
+         "metadata": {
+            "annotations": {
+               "description": "Hot Rod's port."
+            },
+            "labels": {
+               "application": "${APPLICATION_NAME}"
+            },
+            "name": "${APPLICATION_NAME}-hotrod"
+         },
+         "spec": {
+            "ports": [
+               {
+                  "port": 11222,
+                  "targetPort": 11222
+               }
+            ],
+            "selector": {
+               "deploymentConfig": "${APPLICATION_NAME}"
+            }
+         }
+      },
+      {
+         "apiVersion": "v1",
+         "kind": "Service",
```

Comment:
Do we want to expose the Memcached enpoint? It was removed from the other services, AFAIR.

Reply:
We removed the cache, but the service remains exposed in the templates: https://github.com/jboss-container-images/jboss-dataservices-image/blob/dataservices-dev/templates/caching-service.json#L139-L161

I think we should remove the memcached Service from all three templates and if the user wants to configure memcached in the primary store then they can also create said service.

Reply:
I totally agree.

Reply:
PR: https://github.com/jboss-container-images/jboss-dataservices-image/pull/59